addons/udes_stock_routing/models/stock_rule.py

```diff
@@ -0,0 +1,44 @@
+from odoo import api, fields, models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    u_push_on_drop = fields.Boolean(
+        "Push on Drop-off",
+        default=False,
+        help="When stock is dropped in the "
+        "from_location of this path, "
+        "automatically push it onwards.",
+    )
+
+    @api.model
+    def get_path_from_location(self, location):
+        """
+        Find a single stock.rule for which the given location is
+        a valid starting location. If one not found return an empty recordset.
+        """
+        push_steps = self.env["stock.rule"].search(
```

Comment:
sorry, another nit: Rule = self.env["stock.rule"] at the top then do Rule.search... below